### PR TITLE
Product Info

### PR DESCRIPTION
**Pull request by WildMountainAir**
Hoping this pull request shows the commits.

### CODE REVIEW

File: client/src/app.jsx [hunk 1]
```diff
@@ -1,68 +1,61 @@
...
+    });
   }, []);
 
+  if (Object.keys(reviews).length === 0
+  || Object.keys(product).length === 0
+  || Object.keys(styles).length === 0) {
+    return (
+      <div className="loading loading-img">
+        <h1 className="loading-h1">The Proto Company</h1>
+        <h3 className="loading-h3">504: Oops! Please Refresh The Page</h3>
```
Comment (m0nstera) [inline client/src/app.jsx:47]: I like the alternative header!

File: client/src/components/detailedInfo.jsx [hunk 2]
```diff
@@ -1,37 +1,45 @@
 import React from 'react';
-import { Accordion, Card, Button } from 'react-bootstrap';
+import Accordion from 'react-bootstrap/Accordion';
```
Comment (m0nstera) [inline client/src/components/detailedInfo.jsx:2]: Accordion!

File: client/src/components/features.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import Card from 'react-bootstrap/Card';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import { v4 as uuidv4 } from 'uuid';
+
+const Features = ({ product }) => {
+  const featureList = product && product.features;
+  const featureSet = new Set(featureList);
+  const featureArray = [...featureSet];
```
Comment (m0nstera) [inline client/src/components/features.jsx:11]: I appreciate the use of spread operator and a new Set

File: client/src/components/styleNames.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Style and Selected Style
+// How will you display the style name?
+// How will you label the style so users know it's the style?
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const StyleName = ({ styles }) => (
+  <div>
+    <h3 className="styles-name">
+      Style →&nbsp;
```
Comment (m0nstera) [inline client/src/components/styleNames.jsx:10]: Is this little arrow an emoji? Hard to see on Github. Curious what it's pointing to

File: client/src/components/productInfo.jsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+  const getTheAverageRating = (starData) => {
+    console.log('ProductInfo: star data ', starData);
+    let reviewCount = 0;
+    let reviewTotal = 0;
+    const ratingArr = Object.entries(starData);
+    for (let i = 0; i < ratingArr.length; i += 1) {
+      const current = ratingArr[i];
+      const stars = Number(current[0]);
+      const counter = current[1];
```
Comment (m0nstera) [inline client/src/components/productInfo.jsx:31]: Always appreciate your alias'!

**Review by m0nstera — COMMENTED** [review]
Overall really excellent! It's clear that you output really quality work, and you really have some neat things in here.

File: client/src/components/styleNames.jsx (hunk 4 above)
Comment (WildMountainAir) [reply client/src/components/styleNames.jsx:10]: It is an arrow icon and `&nbsp` means a non-breaking space, essentially I needed a single space between the arrow and the category name and didn't want to have to use a string 😆

**Pull request merged**